cpp/include/Ice/Endpoint.h

```diff
@@ -9,41 +9,48 @@
 #include "Ice/Version.h"
 
 #include <string>
-
-#if defined(__clang__)
-#    pragma clang diagnostic push
-#    pragma clang diagnostic ignored "-Wshadow-field-in-constructor"
-#elif defined(__GNUC__)
-#    pragma GCC diagnostic push
-#    pragma GCC diagnostic ignored "-Wshadow"
-#endif
+#include <vector>
 
 namespace Ice
 {
     /**
-     * Base class providing access to the endpoint details.
+     * The user-level interface to an endpoint.
      * \headerfile Ice/Ice.h
      */
-    class ICE_API EndpointInfo
+    class ICE_API Endpoint
```

Comment:
Change the order in this file (and other files): first Endpoint, then EndpointInfo and all other Info classes.